peer/connector.go

```diff
@@ -151,8 +197,30 @@ fail:
 	return errX
 }
 
+// combine multi error into one
+func compositeError(errors []error) error {
+	if nil == errors || 0 == len(errors) {
+		return nil
+	}
+	var ce strings.Builder
+	ce.WriteString("composite error: [")
+	len := len(errors)
+	for i, e := range errors {
+		ce.WriteString(e.Error())
+		if i < len-1 {
+			ce.WriteString(", ")
+		}
+	}
+	ce.WriteString("]")
+	return fault.GenericError(ce.String())
+}
+
 func (conn *connector) allClients(f func(client *upstream.Upstream, e *list.Element)) {
 	for _, client := range conn.staticClients {
+		if nil == client {
```

Comment:
Should we check the nil == client here or in f ? I like the original idea better because, it saids the f function decide what to do with client=nil. I think only f known what does client == nil means